src/commands/register.ts

```diff
@@ -1,46 +1,66 @@
+import { CommandInteraction, Message } from "discord.js";
 import { Db } from "mongodb";
 
-async function subcmd_new(interaction, db: Db) {
-    const gameTag = interaction.options.getString("val_id");
-    const pronouns = interaction.options.getString("pronouns");
-    const result = await db.collection("members").insertOne({
-        gameTag,
-        pronouns,
-        discordId: interaction.user.id,
-    });
+import { MessageExecutable } from "./command";
 
-    if (!result) {
-        interaction.reply({
-        content:
-            "Ah, my eyes are down. I couldn't register you; check with an admin.",
-        ephemeral: true,
-        });
-    } else {
-        interaction.reply({
-        content:
-            "I know **exactly** who you are; you're registered! Feel free to update with /register update",
-        ephemeral: true,
+class NewSubcommand extends MessageExecutable<CommandInteraction> {
```

Comment:
I think this name could be more descriptive, like `NewUserSubcommand` or something along those lines that includes "User" or "Register" somehow

Reply:
Addressed in 2621dee554385cda79c1ab2d33c7350078b13e3f.